### PR TITLE
Added docker build and docker compose file

### PR DESCRIPTION
**Pull request by dkhenry**
vets-api now has a docker file to build a host container and a docker-compose file to run a full stack

### CODE REVIEW

**Comment by robbiethegeek** [comment]
Please update the README.md with instructions on the use / feeding of docker and docker compose, ENV variables and expected paths / locations...

**Comment by dkhenry** [comment]
README.md has been updated.

**Comment by jcmeloni-usds** [comment]
@dkhenry will you please clear up merge conflicts and get this in so we can clean up the PRs?  Thanks! cc @robbiethegeek @ayaleloehr

File: README.md [hunk 1]
```diff
@@ -174,6 +174,19 @@ curl --header "Authorization: Token token=GvmkAW231VxGHkYxyppr2QQsi1D7PStqeiJXyy
 curl --header "Authorization: Token token=GvmkAW231VxGHkYxyppr2QQsi1D7PStqeiJXyyja" localhost:3001/v0/profile
 ```
 
+## Development Setup ( Docker )
```
Comment (omgitsbillryan) [inline README.md]: Am I correct in assuming that the motivation to setup `vets-api` in Docker locally would be for those who don't want to fuss with `vets-api` at all, such as front-end developers? (sorry, I'm no Docker expert).  If so, we should add a link to our existing [Developer Setup](https://github.com/department-of-veterans-affairs/vets-api/blob/cdafd7e052b09e78f4e07518ea7eecbd3cdff1bf/README.md#developer-setup) section that reads, "Use Docker for vets-api" that anchor links to your "Development Setup (Docker)" section.

Comment (dkhenry) [inline README.md]: It is both for developers who don't want to deal with setting it up, but also to ease both the testing and development setup overhead. the compose file create all the necessary infrastructure with a single command ( Postgres, Redis, Specific version of Ruby ) so it reduces the amount of time someone would need to spend preparing to work on the project. 

Then from a operations perspective we can also turn more of our build servers into general purpose builders.

**Comment by dkhenry** [comment]
@jcmeloni-usds @robbiethegeek-usds : This has been updated to head.

**Comment by dkhenry** [comment]
Also I have removed the db_url from the status message.

File: README.md (hunk 1 above)
Comment (omgitsbillryan) [inline README.md]: Gotcha.  So maybe directly under our [Develop Setup](https://github.com/department-of-veterans-affairs/vets-api#developer-setup) you could put something like:
```
As an alternative, you may use [Docker to setup Postgres, Redis, & specific version of Ruby](https://github.com/department-of-veterans-affairs/vets-api/blob/2b9ba822f0e1aaf0fbb7882574ea69e80fd53611/README.md#development-setup--docker-)
```

Comment (dkhenry) [inline README.md]: Good idea. I shall add that in.

**Comment by dkhenry** [comment]
somehow I greatly messed up the commit lineage of this branch. I have rebased everything into a single commit.

File: dk-dockerize.patch [hunk 2]
```diff
@@ -0,0 +1,213 @@
+diff --git a/.dockerignore b/.dockerignore
```
Comment (ayaleloehr) [inline dk-dockerize.patch]: @dkhenry is this file supposed to be in here?

**Comment by dkhenry** [comment]
No :-(

**Comment by dkhenry** [comment]
Removed that file and rebased on master again.

**Comment by robbiethegeek-usds** [comment]
I am getting this setup and will run it, once it is working I will go back and do a real review, but from the cursory review it looks good.

**Comment by dkhenry** [comment]
If we know of any developers who run windows or linux I would love to get a test run on that. This so far has only been tested on OSX

**Comment by robbiethegeek-usds** [comment]
Starting trying to get this off the ground on my VA laptop, on the VA network.  There is an issue with the VPN grabbing all the non-internet routable subnets.  I am working with @dkhenry on this.

**Comment by CyberKoz** [comment]
I think we may be running into an issue like this one 

https://github.com/boot2docker/boot2docker/issues/628

There are some workarounds in that, but I am thinking it might just be easier to use Docker for Mac rather then try to enable a serial console or mess with the routing tables manually.

**Comment by ayaleloehr** [comment]
I believe #1222 closed this out instead and this should be closed. Is that accurate? (and if so, please close).

**Comment by ayaleloehr** [comment]
No one responded to ^^^, so I'm closing.

**Pull request closed without merging**